lib/skate/detours/detours.ex

```diff
@@ -5,23 +5,99 @@ defmodule Skate.Detours.Detours do
 
   import Ecto.Query, warn: false
   alias Skate.Repo
-
+  alias Schedule.Gtfs.Direction
   alias Skate.Detours.Db.Detour
   alias Skate.Settings.User
 
+  @type t :: %__MODULE__{
+          route: String.t(),
+          direction: String.t(),
+          name: String.t(),
+          intersection: String.t(),
+          updated_at: integer(),
+          author_id: integer(),
+          status: :active | :draft | :past
+        }
+
+  @derive Jason.Encoder
+
+  defstruct [
+    :route,
+    :direction,
+    :name,
+    :intersection,
+    :updated_at,
+    :author_id,
+    :status
+  ]
+
   @doc """
-  Returns the list of detours.
+  Returns the detours grouped by active, draft, and past.
 
   ## Examples
 
-      iex> list_detours()
-      [%Detour{}, ...]
-
+      iex> grouped_detours()
+      %{
+        active: [%Detour{}, ...],
+        draft: [%Detour{}, ...],
+        past: [%Detour{}, ...]
+      }
   """
-  def list_detours do
-    Repo.all(Detour)
+  def grouped_detours(user_id) do
+    detours = Detour
+    |> Repo.all()
+    |> Enum.map(&db_detour_to_detour(&1, user_id))
+    |> Enum.group_by(fn detour -> detour.status end)
+
+    %{
+      active: Map.get(detours, :active),
+      draft: Map.get(detours, :draft),
+      past: Map.get(detours, :past)
+    }
   end
 
+  @spec db_detour_to_detour(Detour.t(), integer()) :: t()
+  defp db_detour_to_detour(
+         %{
+           state: %{
+             "context" => %{
+               "route" => %{"name" => route_name},
+               "routePattern" => %{"headsign" => headsign, "directionId" => direction_id}
+             }
+           }
+         } = db_detour,
+         user_id
+       ) do
+    direction = Direction.id_to_string(direction_id)
+
+    date =
+      db_detour.updated_at
+      |> DateTime.from_naive!("Etc/UTC")
+      |> DateTime.to_unix()
+
+    %__MODULE__{
+      route: route_name,
+      direction: direction,
+      name: headsign,
+      intersection: "TBD",
+      updated_at: date,
+      author_id: db_detour.author_id,
+      status: categorize_detour(db_detour, user_id)
+    }
+  end
+
+  @type detour_type :: :active | :draft | :past
+
+  @spec categorize_detour(map(), integer()) :: detour_type
+  defp categorize_detour(%{state: %{"value" => %{"Detour Drawing" => "Active"}}}, _user_id),
+    do: :active
+
+  defp categorize_detour(%{state: %{"value" => %{"Detour Drawing" => "Past"}}}, _user_id),
+    do: :past
+
+  defp categorize_detour(%{author_id: author_id}, user_id) when author_id == user_id, do: :draft
```

Comment:
I do like this usage of Elixir pattern matching - makes it very easy to separate out that drafts are intended to be author-specific, while the other two aren't.